client/src/services/token.ts

```diff
@@ -1,27 +1,18 @@
-import { graphLinks } from '../graphHelper'
-
 //TODO: Make this context rather than class
 export class TokenService {
     private static token = localStorage.getItem('token') ?? ''
-    private static expirationDate = new Date(localStorage.getItem('expirationDate') as string) ?? new Date()
-    static timeToExpire: number = 60 * 60 * 1000 - 100 // 1 hour - 100ms buffer
-
-    public static async getToken() {
-        //console.log('Valid Token', this.expirationDate.getTime() > Date.now())
-        if (!this.isTokenValid()) {
-            this.token = await fetch(graphLinks.token, {
-                method: 'GET',
-            }).then(response => response.text())
-            this.expirationDate = new Date()
-            this.expirationDate.setTime(this.expirationDate.getTime() + this.timeToExpire)
-            localStorage.setItem('token', this.token)
-            localStorage.setItem('expirationDate', this.expirationDate.toString())
-        }
 
+    public static getToken() {
         return this.token
     }
 
-    private static isTokenValid() {
-        return this.token && this.expirationDate.getTime() > Date.now() && !this.token.includes('error')
+    public static async setToken(tokenValue: string) {
+        localStorage.setItem('token', tokenValue)
```

Comment:
FYI, Cookie is necessary mostly because Chrome flag site some times when we try to store token in localstorage

Reply:
Afaik cookie is used for server authentication and localstorage works for client side, do you suggest any feedback here? i am not sure i understand this comment.

Reply:
The thing is we have two auth, Server auth with login, and then MS auth with Token. 
Storing the Token in LocalStorage is not good solution.
Typically The BE handle generating the Token and request data with calls, but since we connecting with MS graph Directly from Client, we need to have to add the token with each call, my idea was to relay on Cookies to do both

Reply:
Makes sense, but since our cookie in backend is httpOnly, it can not be read on client side, you can try to read it using document.cookie but it will not work, hence we need to have it in localStorage

https://www.cookiepro.com/knowledge/httponly-cookie/

Reply:
Agreed, can we have another cookie sent as a regular one ?
Or maybe have that flag temp off ?

Reply:
I would not recommend it due to XSS attacks possibilities, these kind of cookies are not going to be supported for long, but i can say from my experience most apps use hybrid cookie and localStorage approach, cookies are mainly for server to server auth and in my opinion the security point of view should not be ignored :)